src/Sylius/Bundle/UiBundle/Resources/translations/flashes.en.yml

```diff
@@ -28,3 +28,4 @@ sylius:
         update_rules: 'Some rules of the promotions with codes %codes% have been updated.'
     taxon:
         menu_taxon_delete: 'You cannot delete a menu taxon of any channel.'
+        main_taxon_delete: 'You cannot delete this taxon, it is used as main taxon for products with codes %codes%'
```

Comment:
```suggestion
        main_taxon_delete: 'You cannot delete this taxon, it is used as the main taxon for at least products with codes %codes% (max. 20)'
```

Reply:
thanks @lchrusciel sure I will add behat scenario
but `(max. 20)` do you think it adds value?
and if yes, at least it should be dynamic.

Reply:
Let's make it dynamic, that's a good idea. I think so, especially for newcomers. As you wanted to delete taxon and then you received information, that you need to remove it from 20 products. You did it and still the same message, while you could expect, that that's it. In the future, we could add a count of all products.